src/XrdHttp/XrdHttpReq.cc

```diff
@@ -2369,7 +2370,11 @@ int XrdHttpReq::PostProcessHTTPReq(bool final_) {
 
 
         // If we are here it's too late to send a proper error message...
-        if (xrdresp == kXR_error) return -1;
+        if (xrdresp == kXR_error) {
+          XrdHttpError errResp = XrdHttpError::Create(xrdresp, xrderrcode, XrdHttpReq::rtPUT, (XRequestTypes)ntohs(xrdreq.header.requestid), etext, NULL, NULL);
+          errResp.Send(prot, keepalive);
```

Comment:
Unused return value. It should probably be checked (I know it's not checked before, but if the return value doesn't matter, we can make it `void` instead).